path-manager/app/services/PathStore.scala

```diff
@@ -125,4 +125,33 @@ object PathStore {
     val paths = pathItems.map{ PathRecord(_) }
     paths.groupBy(_.`type`)
   }
+
+  def putPathItemAndAwaitIndexUpdate(record: PathRecord) {
+
+    Dynamo.pathsTable.putItem(record.asDynamoItem)
+
+    def waitForIndexUpdate(attempts: Int = 0): Boolean = {
+
+      if (attempts > 20) {
+        false
+      } else {
```

Comment:
should theoretically not happened, but I think it is worth logging an error in that case.